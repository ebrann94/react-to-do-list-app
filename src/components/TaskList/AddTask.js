import React, {useState} from 'react';

const AddTask = ({ handleAddTask }) => {
    const [inputText, setInputText] = useState("");

    return (
        <div className="add-item">
            <form onSubmit={e => {
                e.preventDefault();
                if (inputText) {
                    handleAddTask(inputText);
                    setInputText('');
                }
            }}>
                <input
                    value={inputText}
                    onChange={e => setInputText(e.target.value)}
                    name="addItem"
                    className="add-item__text-input"
                    placeholder="Add a new item..."
                    />
                <button className="add-item__btn" disabled={!inputText}>Add Item</button>
            </form>
        </div>
    );
};

export default AddTask;