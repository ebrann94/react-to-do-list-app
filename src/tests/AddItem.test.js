import React from 'react';
import { shallow, mount } from 'enzyme';
import AddTask from '../components/TaskList/AddTask';

describe('AddTask component tests', () => {
    let wrapper;
    const props = {
        handleAddItem: jest.fn()
    }
    beforeEach(() => {
        wrapper = mount(<AddTask {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('renders without crashing with props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('on input change, sets the state', () => {
        const event = {
            target: {
                value: 'Hello'
            }
        }
        wrapper.find('input').simulate('change', event);
        expect(wrapper.find('input').props().value).toBe('Hello');
    });

    it('calls handler function on submit', () => {
        const event = {
            target: {
                value: 'Washing Up'
            }
        }
        wrapper.find('input').simulate('change', event);
        wrapper.find('form').simulate('submit', { preventDefault() {} });
        expect(props.handleAddItem).toHaveBeenCalledWith(event.target.value);
    });
});