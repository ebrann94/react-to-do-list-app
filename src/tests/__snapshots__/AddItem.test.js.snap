// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItem component tests renders without crashing with props 1`] = `ReactWrapper {}`;
